apps/er-coap/er-coap.c

```diff
@@ -123,7 +123,7 @@ coap_set_option_header(unsigned int delta, size_t length, uint8_t *buffer)
     } else if(*x > 12) {
       buffer[++written] = (*x - 13);
     }
-  } while(x != &length && (x = &length));
+  } while(x != (unsigned int*)&length && (x = (unsigned int*)&length));
 
```

Comment:
Sounds better, let the compiler optimize. Are you sure about the type
uint16_t instead of unsigned?
On Sep 9, 2015 2:57 PM, "Moritz Strübe" <notifications@github.com> wrote:

> In apps/er-coap/er-coap.c
> <https://github.com/contiki-os/contiki/pull/1250#discussion_r39038096>:
>
> > @@ -123,7 +123,7 @@ coap_set_option_header(unsigned int delta, size_t length, uint8_t *buffer)
> >      } else if(*x > 12) {
> >        buffer[++written] = (*x - 13);
> >      }
> > -  } while(x != &length && (x = &length));
> > +  } while(x != (unsigned int*)&length && (x = (unsigned int*)&length));
> >
>
> Yea found that out by now, too. :( What about rewriting it to
>
>     uint16_t ptr[] = {&size, $length)
>     int pos;
>     for(pos = 0; pos < 2; pos++){
>        ptr[pos].....
>
> The Compiler should be able to optimize it or even unroll.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/contiki-os/contiki/pull/1250/files#r39038096>.
>